internal/utils/config.go

```diff
@@ -53,56 +45,60 @@ func ValidateConfig(config *TomlConfig) (*Config, error) {
 			tenant.DatabaseUrl = &databaseUrl
 		}
 
-		if tenant.DatabaseUrl != nil && config.TenantDatabaseGroupAuth != nil {
-			databaseUrl := *tenant.DatabaseUrl + "?auth_token=" + *config.TenantDatabaseGroupAuth
-			tenant.DatabaseUrl = &databaseUrl
+		if tenant.DatabaseUrl == nil {
+			return nil, fmt.Errorf("utils: config needs to define either \"[tenants.%s].database_url\" or \"tenant_database_url_template\"", key)
 		}
 
 		config.Tenants[key] = tenant
 	}
 
-	shouldUseSuperDatabase := false
-	isMultiTenant := false
-
+	configTenantsSize := len(config.Tenants)
 
-	var tenantDatabaseUrlTemplate *string
-	tenantDatabaseUrlTemplate = nil
+	if configTenantsSize == 0 && config.SuperDatabaseUrl == nil {
+		return nil, fmt.Errorf("utils: config needs to define either \"[tenants.*]\" or \"super_database_url\"")
+	}
 
-	if superDatabaseUrl != nil && config.Tenants == nil {
-		shouldUseSuperDatabase = true
-		isMultiTenant = true
-		if (config.TenantDatabaseUrlTemplate != nil) {
-			tenantDatabaseUrlTemplate = config.TenantDatabaseUrlTemplate
+	if configTenantsSize > 0 {
+		if config.SuperDatabaseUrl != nil {
+			// TODO: logger ? after echo starts ?
+			fmt.Printf("utils: using %d config \"[tenants.*]\", ignoring config \"super_database_url\"\n", configTenantsSize)
 		}
-		if (config.TenantDatabaseUrlTemplate != nil && config.TenantDatabaseGroupAuth != nil) {
-			newTenantDatabaseUrlTemplate := *tenantDatabaseUrlTemplate + "?auth_token=" + *config.TenantDatabaseGroupAuth
-			tenantDatabaseUrlTemplate = &newTenantDatabaseUrlTemplate
-		}
-	} else {
-		superDatabaseUrl = nil
-	}
 
+		return &Config{
+			IsMultiTenant:             configTenantsSize > 1,
+			ShouldUseSuperDatabase:    false,
+			SuperDatabaseUrl:          nil,
+			TenantDatabaseUrlTemplate: config.TenantDatabaseUrlTemplate,
+			Tenants:                   config.Tenants,
+		}, nil
+	}
```

Comment:
Since previous code ignored superDatabaseUrl if tenants were set, modified to an early return.